aws-athena/views/reporting-vw_assessment_roll_muni.sql

```diff
@@ -77,42 +77,92 @@ pin_counts AS (
         vptc.year,
         vptc.major_class,
         stages.stage
+),
+
+/* This CTE solely provides lagged values by year and stage by PIN. Because
+parcel universes change year-to-year (PINs are activated/deactivated, change
+class), building "prior phase" stats by aggregating lagged values will lead to
+small disgareements between a given year's "prior phase" stats for its mailed
+stage and the prior year's actual bor certified stats. */
+vpvl AS (
+
+    SELECT
+        *,
+        LAG(tot)
+            OVER (
+                PARTITION BY
+                    pin
+                ORDER BY year ASC, stage_num ASC
+            )
+            AS tot_lag
+    FROM {{ ref('reporting.vw_pin_value_long') }}
+
+),
+
+/* Calculate total and median values by municipality, as well as the portion of
+each municipality that has progressed through an assessment stage by class. This
+is the meat of the view but needs to be a CTE since the view contains columns
+aggregated by different groupings. */
+muni_aggregated AS (
+    SELECT
+        pin_counts.year,
+        pin_counts.stage,
+        CASE
+            WHEN pin_counts.stage = 'mailed' THEN 1
+            WHEN pin_counts.stage = 'assessor certified' THEN 2
+            WHEN pin_counts.stage = 'bor certified' THEN 3
+        END AS stage_num,
```

Comment:
**nitpick:** Why not do this in the CTE so it can be used by both the select and group by?